common.go

```diff
@@ -81,35 +84,60 @@ func detachedSignatureInputFromHash(plaintextAndHeaderHash []byte) []byte {
 	return buf.Bytes()
 }
 
-func hmacSHA512256(key []byte, input []byte) []byte {
-	// Equivalent to crypto_auth, but using Go's builtin HMAC. Truncates
-	// SHA512, instead of actually calling SHA512/256.
-	if len(key) != cryptoAuthKeyBytes {
-		panic("Bad crypto_auth key length")
+func copyEqualSize(out, in []byte) {
+	if len(out) != len(in) {
+		panic(fmt.Sprintf("len(out)=%d != len(in)=%d", len(out), len(in)))
 	}
-	authenticatorDigest := hmac.New(sha512.New, key)
-	authenticatorDigest.Write(input)
+	copy(out, in)
+}
+
+func copyEqualSizeStr(out []byte, in string) {
```

Comment:
Do I have it right, that the main benefit here is avoiding the implicit alloc that would come with a `[]byte(_)` cast?